src/mqtt/mqtt_test.c

```diff
@@ -988,11 +988,6 @@ TEST( MqttTest, MQTT_Subscribe_Publish_With_Qos_1 )
             /* Timeout. */
             break;
         }
-        else if( receivedPubAck != 0 )
-        {
-            /* Received the SUBACK. No need to call process loop anymore. */
-            break;
-        }
```

Comment:
This had to be removed as this loop is also supposed to wait for the incoming publish and not just the PUBACK.